### PR TITLE
chore(pdm): migrate from pipenv to pdm

### PR DESCRIPTION
**Pull request by fnareoh**
I had a bit of time so I though I could start migrating pygitguardian to pdm :) 

~~For now I have a version that passes checks but I duplicated section from [the setup.py ](https://github.com/GitGuardian/py-gitguardian/blob/master/setup.py#L35) which is probably not the way to go... Corrections are very welcome...~~

Should be good now !

### CODE REVIEW

**Comment by deronnax** [comment]
Nop indeed. You should import the setup.py with `pdm import setup.py`, merge and unduplicate the added information in the pyproject.toml, and delete the setup.py.

File: pyproject.toml [hunk 1]
```diff
@@ -17,4 +85,4 @@ reportUnnecessaryTypeIgnoreComment = true
 version = "literal: pygitguardian/__init__.py: __version__"
 format = "md"
 md_header_level = "2"
-insert_marker = "# Changelog"
+insert_marker = "# Changelog"
```
Comment (agateau-gg) [inline pyproject.toml:88]: Looks like your editor is not configured to write trailing new lines (same thing for setup.cfg).

File: pdm.lock
Comment (agateau-gg) [inline pdm.lock:1]: Since py-gitguardian is a lib, it should not have a pdm.lock: only apps need one.

**Review by agateau-gg — CHANGES_REQUESTED** [review]
Super happy that you took the time to work on this 🤩.

Just made a few minor remarks.

File: pyproject.toml [hunk 2]
```diff
@@ -1,3 +1,71 @@
...
+]
+
+[project.urls]
+Homepage = "https://github.com/GitGuardian/py-gitguardian"
+
+[tool.pdm]
+[[tool.pdm.source]]
+name = "pypi"
+url = "https://pypi.python.org/simple"
+verify_ssl = true
```
Comment (agateau-gg) [inline pyproject.toml:47]: You can remove this part: it's already there by default.

File: pyproject.toml [hunk 3]
```diff
@@ -1,3 +1,71 @@
...
+description = "Python Wrapper for GitGuardian's API -- Scan security policy breaks everywhere"
+keywords = [
+    "api-client devsecops secrets-detection security-tools library gitguardian",
+]
+readme = "README.md"
+authors = [
+    {name = "GitGuardian", email = "support@gitguardian.com"},
+]
+maintainers = [
+    {name = "GitGuardian"},
```
Comment (agateau-gg) [inline pyproject.toml:13]:
```suggestion
    {name = "GitGuardian", email = "support@gitguardian.com"},
```

**Review by agateau-gg — CHANGES_REQUESTED** [review]
A few other tiny issues I missed.

**Review by agateau-gg — APPROVED** [review]
🎉

**Pull request merged**